e2e-tests/test_authentication.py

```diff
@@ -0,0 +1,104 @@
+        cls.create_from_yaml(cls, f"{cls.tempdir}/vapi-auth-service.yaml")
+        assert wait_until_ready(label_selector="app=vapi-auth")
+
+    def test_http_basic_authentication(self):
+        """Test rapidast with HTTP Basic authentication configured"""
+
+        self.create_from_yaml(f"{self.tempdir}/rapidast-vapi-configmap-http-basic.yaml")
+        self.create_from_yaml(f"{self.tempdir}/rapidast-vapi-pod-http-basic.yaml")
+        assert is_pod_with_field_selector_successfully_completed(
+            field_selector="metadata.name=rapidast-vapi-http-basic", timeout=360
+        )
+
+        logs = get_log_from_pod(self.tempdir, "rapidast-vapi-http-basic", container="rapidast", log_format="text")
+        data = get_log_from_pod(
+            self.tempdir,
+            "rapidast-vapi-http-basic",
+            filename_suffix="results",
+            container="results",
+            log_format="json",
+        )
+
+        # Verify that HTTP Basic authentication was configured correctly in logs
+        assert (
+            "ZAP configured with HTTP Basic Authentication" in logs
+        ), "ZAP logs should indicate HTTP Basic authentication was configured"
+
+        # Verify that the Authorization Basic header with correct credentials is present
+        # NOTE: "user:mypassw0rd" are dummy test credentials for e2e testing - not real secrets
+        expected_credentials = base64.b64encode(b"user:mypassw0rd").decode("utf-8")
+        basic_auth_header_found = verify_specific_auth_header_value(
+            data, "Authorization", f"Basic {expected_credentials}"
+        )
+        assert (
+            basic_auth_header_found
+        ), "Authorization header with correct Basic credentials should be found in scan results"
+
+    def test_http_header_authentication(self):
+        """Test rapidast with HTTP Header authentication configured"""
+
+        self.create_from_yaml(f"{self.tempdir}/rapidast-vapi-configmap-http-header.yaml")
+        self.create_from_yaml(f"{self.tempdir}/rapidast-vapi-pod-http-header.yaml")
+        assert is_pod_with_field_selector_successfully_completed(
+            field_selector="metadata.name=rapidast-vapi-http-header", timeout=360
+        )
+
+        logs = get_log_from_pod(self.tempdir, "rapidast-vapi-http-header", container="rapidast", log_format="text")
+        data = get_log_from_pod(
+            self.tempdir,
+            "rapidast-vapi-http-header",
+            filename_suffix="results",
+            container="results",
+            log_format="json",
+        )
+
+        assert (
+            "ZAP configured with Authentication using HTTP Header" in logs
+        ), "ZAP logs should indicate HTTP Header authentication was configured"
+
+        # NOTE: "MySecretHeader" is a dummy test header value for e2e testing - not a real secret
+        custom_header_found = verify_specific_auth_header_value(data, "Authorization", "MySecretHeader")
```

Comment:
Just to clarify, these tests are just checking that ZAP logged use of the expected auth headers, not that ZAP could authenticate to the VAPI instance (I didn't see any auth config for VAPI)?

Reply:
Exactly, this test just checks that ZAP sends the expected auth headers. We could make it stricter (like checking if a specific ZAP rule about clear-text auth shows up), but I went with something simpler. The point isn’t to test if auth works in VAPI, just to confirm that ZAP is actually sending the headers